src/Relation/Binary/Construct/SetoidFromPartialSetoid.agda

```diff
@@ -0,0 +1,58 @@
+------------------------------------------------------------------------
+-- The Agda standard library
+--
+-- Conversion of a PartialSetoid into a Setoid
+------------------------------------------------------------------------
+
+open import Relation.Binary.Bundles using (PartialSetoid; Setoid)
+
+module Relation.Binary.Construct.SetoidFromPartialSetoid
+  {a ℓ} (S : PartialSetoid a ℓ) where
+
+open import Function.Base using (id; _on_)
+open import Level using (_⊔_)
+open import Relation.Binary.Core using (Rel)
+open import Relation.Binary.Structures
+  using (IsEquivalence)
+open import Relation.Binary.Morphism.Structures
+  using (IsRelHomomorphism; IsRelMonomorphism)
+
+private
+  module S = PartialSetoid S
+
+
+------------------------------------------------------------------------
+-- Definitions
+
+record Carrier : Set (a ⊔ ℓ) where
```

Comment:
I must say that I expected the carrier to be called `Carrier` (and not `ι`) with the record called something like 'Reflective' or some such. 

Thinking some more, I think having a definition for `Defined x` (expanding to `x S.≈ x`) might make the overall definition more perspicuous.

Reply:
Good realization - I certainly had not spotted that. Perhaps sitting is indeed wise.